docs/ContinuousIntegration.md

```diff
@@ -0,0 +1,27 @@
+# Continuous Integration
+
+This repo makes use of GitHub Actions to run various checks to aid maintaining this repo. Actions allow us to run certain checks on PRs, maintain our dependencies, and have automatic issue checking. All code for workflows and actions can be found in the `.github` directory at the root of this repository.
+
+## Automatic tests
+
+To make sure that master branch is alway safe, all of our tests for the UI are run on each pull request. For full details on our tests, [view our testing documentation](./Test.md). The output of the tests run by Jest can be viewed in the logs of the action on a pull request. For E2E tests, the results are also available in the logs, and the failure output will be uploaded as an artifact on the action. These can be downloaded via the `Artifacts` dropdown at the top right of the action page.
+
+### Test coverage information
+
+As part of automatically running the tests, if they fail because of insufficient test coverage, the coverage report from the tests are formatted as a report on the pull request. This way, it is easy to see where the coverage requirements have not been met and is a quick way to let a contributer know what needs to be done. If the tests pass, no report will be given.
+
+## Bundle size calculation
+
+It is important that we maintain as small as a bundle size as possible. To help with this, a bundle size report will be made on a pull request that shows the new size of the bundle and the percentage increase/decrease in size. By having this report, maintainers can easily raise questions if the size of the bundle changes dramatically.
```

Comment:
It would be nice to be able to watch the 'drift' of bundle size across time, as it may not go up dramatically on a single PR, but may do across multiple PRs/releases. Does anyone know of anything that can easily chart these sort of metrics?

Reply:
Absolutley agree. There are plenty of plugins and tools which could be used from 3rd parties for this - from the ones I have seen they normally require elevated priviledges in github/payment however. We could however in an action build `master`, `master at last release` (etc), and `this`, branch, and compare the generated bundle JSON to get a similar effect.